lib/utils.c

```diff
@@ -39,6 +39,54 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[41], buf[80], *endp;
+	ssize_t len;
+	FILE *fp;
+	long result;
+
+	sprintf(fname, "/sys/class/net/%s/%s", dev, prop);
```

Comment:
Let's make this `sprintf()` safer by changing it to `snprintf()`, with the second parameter being `sizeof(fname)`.

Reply:
Let's also catch any errors that happen from `snprintf()` like we did in `parse_percent_rate()`.